test/org.jkiss.dbeaver.test.platform/src/org/jkiss/dbeaver/model/sql/parser/SQLScriptParserTest.java

```diff
@@ -502,6 +501,41 @@ public void parseCurrentControlCommandsCursorTail() throws DBException {
     	SQLScriptElement element = SQLScriptParser.parseQuery(context, 0, query.length(), 15, false, false);
     	Assert.assertEquals("@set col1 = '1'", element.getText());
     }
+
+    @Test
+    public void parseQueryWithCommentsBeforeBlockDeclaration() throws DBException {
+        {
+            String query = "/* Issue */\n" + "DO BEGIN\n" + "SELECT * FROM dummy;\n" + "END;";
+            SQLParserContext context = createParserContext(setDialect("sap_hana"), query);
+            SQLScriptElement element = SQLScriptParser.parseQuery(context, 0, query.length(), 0, false, false);
+            Assert.assertEquals("DO BEGIN\n" + "SELECT * FROM dummy;\n" + "END;", element.getText());
+        }
+        {
+            String query = "/* Issue */" + "DO BEGIN\n" + "SELECT * FROM dummy;\n" + "END;";
+            SQLParserContext context = createParserContext(setDialect("sap_hana"), query);
+            SQLScriptElement element = SQLScriptParser.parseQuery(context, 0, query.length(), 0, false, false);
+            Assert.assertEquals("DO BEGIN\n" + "SELECT * FROM dummy;\n" + "END;", element.getText());
+        }
+        {
+            String query = "/* Issue */\n\n" + "DO BEGIN\n" + "SELECT * FROM dummy;\n" + "END;";
+            SQLParserContext context = createParserContext(setDialect("sap_hana"), query);
+            SQLScriptElement element = SQLScriptParser.parseQuery(context, 0, query.length(), 0, false, false);
+            Assert.assertEquals("DO BEGIN\n" + "SELECT * FROM dummy;\n" + "END;", element.getText());
+        }
+        {
+            String query = "DO BEGIN\n" + "SELECT * FROM dummy;\n" + "END;";
+            SQLParserContext context = createParserContext(setDialect("sap_hana"), query);
+            SQLScriptElement element = SQLScriptParser.parseQuery(context, 0, query.length(), 0, false, false);
+            Assert.assertEquals("DO BEGIN\n" + "SELECT * FROM dummy;\n" + "END;", element.getText());
+        }
+        {
+            String query = "DO BEGIN\n" + "SELECT * FROM dummy;\n" + "END;";
```

Comment:
This test looks exactly like the previous one.

Also, please add more tests not with the Hana dialect.